backends-velox/src/main/scala/org/apache/gluten/extension/CollectRewriteRule.scala

```diff
@@ -48,6 +51,19 @@ case class CollectRewriteRule(spark: SparkSession) extends Rule[LogicalPlan] {
     out
   }
 
+  private def needReplace(plan: LogicalPlan): Boolean = plan match {
+    // Do not replace with objectHashAggregate if disable objectHashAgg collectRewrite.
+    case PhysicalAggregation(_, aggregateExpr, _, _)
+        if !GlutenConfig.getConf.veloxObjectHashAggCollectRewriteEnabled =>
+      val aggregateExpressions = aggregateExpr.map(expr => expr.asInstanceOf[AggregateExpression])
+      val useHash = SparkShimLoader.getSparkShims.supportsHashAggregate(
+        aggregateExpressions.flatMap(_.aggregateFunction.aggBufferAttributes))
+      val useObjectHash = plan.conf.useObjectHashAggregation &&
+        SparkShimLoader.getSparkShims.supportsObjectHashAggregate(aggregateExpressions)
+      useHash || !useObjectHash
+    case _ => true
+  }
```

Comment:
Would you like to explain a bit about this code? Thanks.

Reply:
Updated

Reply:
Sorry I still don't quite understand (thought it will be a simple switch to turn the rule on/off). If all you wanted is to fallback to vanilla Spark's object hash aggregate, why not just disable this rule?

If it's hard to explain, would you like to share the comparison of query plans in text before and after the change?